include/seqan3/alignment/pairwise/policy/alignment_matrix_policy.hpp

```diff
@@ -58,35 +62,35 @@ class alignment_matrix_policy
     //!}
 
     /*!\brief Allocates the memory of the underlying matrices.
-     * \tparam fst_range_t The type of the first sequence to align; must model std::forward_ranges.
-     * \tparam snd_range_t The type of the second sequence to align; must model std::forward_ranges.
-     * \tparam fst_range_t The type of the first sequence to align.
-     * \tparam sec_range_t The type of the second sequence to align.
-     * \param[in] fst_range The first sequence to align.
-     * \param[in] snd_range The second sequence to align.
+     * \tparam fst_sequence_t The type of the first sequence to align; must model std::forward_ranges.
+     * \tparam snd_sequence_t The type of the second sequence to align; must model std::forward_ranges.
+     *
+     * \param[in] fst_sequence The first sequence to align.
+     * \param[in] snd_sequence The second sequence to align.
      *
      * \details
      *
      * Initialises the underlying score and trace matrices and sets the respective matrix iterators to the begin of the
      * corresponding matrix.
      */
-    template <typename fst_range_t, typename snd_range_t>
-    constexpr void allocate_matrix(fst_range_t && fst_range, snd_range_t && snd_range)
+    template <typename fst_sequence_t, typename snd_sequence_t>
```

Comment:
like above we need to make sure that the public interfaces of a class are constrained since they are called with user input. But within the class's private scope these invariants should be still valid. Unless explicitly changed, but then again this was part of a private interface and the class should be well aware above its own invariants.